test_whole_board/test_whole_board.py

```diff
@@ -0,0 +1,210 @@
+# Copyright (c) 2022 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import spynnaker8 as sim
+from spinnaker_testbase import BaseTestCase
+from spalloc.job import Job
+from spalloc.states import JobState
+import pytest
+import tempfile
+import os
+
+
+class WholeBoardTest(object):
+
+    up = [(0, 0), (0, 1), (0, 2), (0, 3),
+          (1, 0), (1, 1), (1, 2), (1, 3), (1, 4),
+          (2, 0), (2, 1), (2, 2), (2, 3), (2, 4), (2, 5),
+          (3, 0), (3, 1), (3, 2), (3, 3), (3, 4), (3, 5), (3, 6),
+          (4, 0), (4, 1), (4, 2), (4, 3), (4, 4), (4, 5), (4, 6), (4, 7),
+          (5, 1), (5, 2), (5, 3), (5, 4), (5, 5), (5, 6), (5, 7),
+          (6, 2), (6, 3), (6, 4), (6, 5), (6, 6), (6, 7),
+          (7, 3), (7, 4), (7, 5), (7, 6), (7, 7)]
+    down = list(up)
+    down.reverse()
+    right = [(0, 0), (1, 0), (2, 0), (3, 0), (4, 0),
+             (0, 1), (1, 1), (2, 1), (3, 1), (4, 1), (5, 1),
+             (0, 2), (1, 2), (2, 2), (3, 2), (4, 2), (5, 2), (6, 2),
+             (0, 3), (1, 3), (2, 3), (3, 3), (4, 3), (5, 3), (6, 3), (7, 3),
+             (1, 4), (2, 4), (3, 4), (4, 4), (5, 4), (6, 4), (7, 4),
+             (2, 5), (3, 5), (4, 5), (5, 5), (6, 5), (7, 5),
+             (3, 6), (4, 6), (5, 6), (6, 6), (7, 6),
+             (4, 7), (5, 7), (6, 7), (7, 7)]
+    left = list(right)
+    left.reverse()
+    ur = [(0, 0), (0, 1), (0, 2), (0, 3),
+          (1, 0), (1, 1), (1, 2), (1, 3), (1, 4),
+          (2, 0), (2, 1), (2, 2), (2, 3), (2, 4), (2, 5),
+          (3, 0), (3, 1), (3, 2), (3, 3), (3, 4), (3, 5), (3, 6),
+          (4, 0), (4, 1), (4, 2), (4, 3), (4, 4), (4, 5), (4, 6), (4, 7),
+          (5, 1), (5, 2), (5, 3), (5, 4), (5, 5), (5, 6), (5, 7),
+          (6, 2), (6, 3), (6, 4), (6, 5), (6, 6), (6, 7),
+          (7, 3), (7, 4), (7, 5), (7, 6), (7, 7)]
+    dl = list(ur)
+    dl.reverse()
```

Comment:
I'm a bit confused as to what the difference is between "up" and "ur" (I assume short for "up right")?  I don't see how the code leads to different chains when using "ur" compared to "up" - is it just that once you've gone up, down, left, right, then you can only next go "up right" and then "down left", and so it picks the correct next chip from the list?

Reply:
(Comment made before "ur" was changed to its current state...)

Reply:
I changed this at almost exactly the time you commented!

Reply:
Testing done manually and the example still works with the given changes

Reply:
The Chips are added to the synfire chain in the order in the list. So up goes from (0, 0) to (0,1) which is in the up position. ect.  UR goes from (0,0) to (1, 4) which is over the up right link.

note: I dont check if the link exists just the chip. If the link os down the router will find an alternative route and not break the chain.

Reply:
The issue was that in the original version, up == ur (see 56fc4f3fba349177cadb4695b91d2a063cfd553b)

Reply:
In reality, the network run here is unimportant.  The only important bits are that it runs on boards that work and produces the expected results, and that it actually runs neural network code on all cores to test that they work when used with that code.